airflow/jobs/scheduler_job.py

```diff
@@ -1632,7 +1633,9 @@ def _orphan_unreferenced_datasets(self, session: Session = NEW_SESSION) -> None:
                     func.count(TaskOutletDatasetReference.dag_id) == 0,
                 )
             )
+            .all()
```

Comment:
This may be a bit resource-intensive. We could do this instead:

```python
def set_orphaned(dataset: DatasetModel) -> int:
    self.log.info("Orphaning unreferenced dataset '%s'", dataset.uri)
    dataset.is_orphaned = expression.true()
    return 1

updated_count = sum(set_orphaned(dataset) for dataset in orphaned_dataset_query)
Stats.gauge("dataset.orphaned", updated_count)
```